sound/soc/sof/ipc4-pcm.c

```diff
@@ -98,8 +98,23 @@ static int sof_ipc4_trigger_pipelines(struct snd_soc_component *component,
 	for (i = pipeline_list->count - 1; i >= 0; i--) {
 		pipe_widget = pipeline_list->pipe_widgets[i];
 		pipeline = pipe_widget->private;
-		if (pipeline->state != state && !pipeline->skip_during_fe_trigger)
-			data->pipeline_ids[data->count++] = pipe_widget->instance_id;
+
+		if (pipeline->skip_during_fe_trigger)
+			continue;
+
+		switch (state) {
+		case SOF_IPC4_PIPE_RUNNING:
+		case SOF_IPC4_PIPE_RESET:
+			/* Do not RESET or repeat RUNNING if the pipeline is in use */
+			if (!pipe_widget->trigger_count)
+				data->pipeline_ids[data->count++] = pipe_widget->instance_id;
+			break;
+		default:
```

Comment:
Should we specific the cases instead of using `default`? From the comment below, only `SOF_IPC4_PIPE_PAUSED` should be set?